app/gui/src/controller/graph/executed.rs

```diff
@@ -267,14 +274,22 @@ impl Handle {
         if self.project.read_only() {
             Err(ReadOnly.into())
         } else {
+            let mut successful_pop_count = 0;
+            let mut result = Ok(());
             for _ in 0..frame_count {
-                self.execution_ctx.pop().await?;
+                if let Err(error) = self.execution_ctx.pop().await {
+                    result = Err(error);
+                    break;
+                }
+                successful_pop_count += 1;
             }
```

Comment:
I would rewrite it to `take_while(...).count()` so we avoid early returns from loop and one mutable variable.

Reply:
Discussed it in private with @farmaazon and rewrote it using the `Stream`/`TryStream` API.